src/main/java/seedu/address/model/task/TimeInterval.java

```diff
@@ -25,7 +28,7 @@
     
     public final String intervalString;
     public final int length;
-    public final String unit;
+    public String unit;
```

Comment:
Not a good idea to remove the final here, is it possible to modify the constructor so unit can still be final? Say by using a temp variable